packages/overlays/src/LocalOverlayController.js

```diff
@@ -1,36 +1,77 @@
 import { render, html } from '@lion/core';
-import { managePosition } from './utils/manage-position.js';
 import { containFocus } from './utils/contain-focus.js';
 import { keyCodes } from './utils/key-codes.js';
 
 export class LocalOverlayController {
   constructor(params = {}) {
-    const finalParams = {
+    const defaultOptions = {
       placement: 'top',
-      position: 'absolute',
-      ...params,
+      positionFixed: false,
+      modifiers: {
+        keepTogether: {
+          enabled: false,
+        },
+        preventOverflow: {
+          enabled: true,
+          boundariesElement: 'viewport',
+          padding: 16, // viewport-margin for shifting/sliding
+        },
+        flip: {
+          boundariesElement: 'viewport',
+          padding: 16, // viewport-margin for flipping
+        },
+        offset: {
+          enabled: true,
+          offset: `0, 8px`, // horizontal and vertical margin (distance between popper and referenceElement)
+        },
+        arrow: {
+          enabled: false,
+        },
+      },
     };
-    this.hidesOnEsc = finalParams.hidesOnEsc;
-    this.hidesOnOutsideClick = finalParams.hidesOnOutsideClick;
-    this.trapsKeyboardFocus = finalParams.trapsKeyboardFocus;
-    this.placement = finalParams.placement;
-    this.position = finalParams.position;
+
+    const finalPlacement = {
+      ...defaultOptions,
+      ...params.placementConfig,
+    };
+
+    let finalModifiers = defaultOptions.modifiers;
+    if (params.placementConfig) {
+      finalModifiers = {
+        ...finalModifiers,
+        ...params.placementConfig.modifiers,
+      };
+    }
+
+    this.placementConfig = {
+      placement: finalPlacement.placement,
+      positionFixed: finalPlacement.positionFixed,
+      modifiers: finalModifiers,
+    };
+
+    this.__preloadPopper();
```

Comment:
I don't think it's a good idea to eagerly preload like this, there is no guarantee this will be run after the application has fully loaded so you're still blocking the initial load. It's better to let applications control this behavior, perhaps with some central manager within lion which can do other preloading too.

Reply:
It's indeed a good idea to get more fine grained control, so we can allow the app developer to decide whether it should compete with other lazily loaded resources.
I assume the first meaningful paint depends on sync resources only or not @LarsDenBakker ?

Reply:
Depends on the definition of meaningful. A banking application is useless until you fetch the account data.

Unfortunately there is no programmatic hook in the platform for determining these landmarks, though I think they will add these things over time.

We could also considering using a prefetch link instead of importing it from the constructor. Prefetch tells the browser you will use a resource at some point, and the browser will fetch it when it's not busy with other things. It will not parse/run the javascript, so it's never blocking the main thread.

Reply:
Good point, but then we do indeed need this generic lion-preloader manager for resource control before I can remove it here.

I'll look into prefetch though as an improvement for the current situation :)

Reply:
For now we will merge it as is, but we'll make sure to look into smarter way to fetch/load this (preloader manager, prefetch, etc.)

Reply:
Then I think you shouldn't preload it in the constructor, and only on use for now. otherwise you're introducing an extra load to the inital load of apps with this.

Reply:
Loading 6k on demand on a slow mobile connection? That would lead to  bad UX when you expect something to happen on a button press. 
Wouldn't it be possible to lazy load the LocalOverlay in your app for now?

Reply:
So will loading extra 6k on startup, right? I can't control how everyone uses the local overlay, so I can't lazy load it in that way.

Reply:
By the time the users presses the invoker button, the 6k should have loaded, so the user experience is just smooth there. 

For now we could say responsibility of every individual feature to be performant. 

Or maybe you could give an example for prefetching dynamic imports?
This shows how to do it for webpack builds at least:
https://medium.com/front-end-weekly/webpack-and-dynamic-imports-doing-it-right-72549ff49234
But that's not what you had in mind?